Feat/register reward partner

Figaarillo opened the pull request.

Figaarillo committed:
feat: create payloads to reward partnew

Figaarillo committed:
feat: create entity to reward partnew

Figaarillo committed:
chore: rename entity class and paylaods

Figaarillo committed:
feat: create repository to reward partner

Figaarillo committed:
feat: create use case to register a reward partner

Figaarillo committed:
feat: create use case to update a reward partner

Figaarillo committed:
chore: change type of cuit

Figaarillo committed:
feat: create dto type

Figaarillo committed:
feat: create dtos to register and update

Figaarillo committed:
feat: create middleware to validate schemas and add little fix in dtos

Figaarillo committed:
feat: create handler to reward partner

Figaarillo committed:
feat: implements repository

Figaarillo committed:
feat: create swagger schema

Figaarillo committed:
feat: create route

Figaarillo committed:
feat: create bootstrap

Figaarillo committed:
feat: register bootstarp function of reward partner

wink15 commented on src/reward-partner/domain/payloads/reward-partner.payload.ts:
export interface LocalAdherido {
  id: string
  name: string
  address: string
  cuit: string
  email: string
  password: string
  username: string
  phoneNumber: string
}

Figaarillo committed:
fix: correct cuit

 - change type of cuit from number to string
 - update migrations
 - update swagger schema

Figaarillo committed:
chore: change DTO to address

Figaarillo committed:
chore: correct DTO of cuit

Figaarillo committed:
refactor: add two atributes more, username and phone number

Figaarillo committed:
fix: fix migrations
